vignettes/epichains.Rmd

```diff
@@ -29,6 +29,38 @@ knitr::opts_chunk$set(
 )
 ```
 
+This vignette demonstrates how get started with using the _epichains_ package
+for simulating transmission chains or estimating the likelihood of observing
+a transmission chain.
+
+::: {.alert .alert-info}
+* To understand the theoretical background of the branching processes methods
+used here, refer to the [theory vignette](https://epiverse-trace.github.io/epichains/articles/theoretical_background.html)
+* To understand the software development design decisions and implementation details of the package,
+refer to the [design vignette](https://epiverse-trace.github.io/epichains/articles/design-principles.html)
+:::
+
+Epidemics spread through when infected individuals transmit the
+infection to others. If we assume that each case infects a random number of
+other individuals (the offspring) through time units called generations,
+and that this transmission process is governed by a reproduction number and
+a probability distribution describing how the offspring are produced through
+the time unit, then the epidemic can be modelled as a branching process.
+
+Branching processes are a stochastic process where each individual in a
+generation gives rise to a random number of individuals in the next generation.
+The distribution of the number of offspring is called the offspring
+distribution.
+
+The size of the transmission chain is the total number of
+individuals infected by a single case, and the length of the transmission
+chain is the number of generations from the first case to the last case they
+produced before the chain ended (See figure below).
+
+```{r out.width = "75%", echo = FALSE, fig.cap = "An example of a transmission chain starting with a single case C1. Cases are represented by blue circles and arrows indicate who infected whom. The chain grows through generations G1, G2, and G3, producing cases C2, C3, C4, C5, C6, and C7. The chain ends at generation G3 with case C7. The size of C1's chain is 7 (sum of all blue circles) and the length is 3 (maximum number of generations reached by C1's chain).", fig.alt = "The figure shows an example of a transmission chain starting with a single case C1. The figure is depicted as blue circles linked by orange directed arrows showing the cases they produced. Each generation is marked by a brown rectangle and labelled Gen 1, Gen 2, and Gen 2. The chain grows through generations G1, G2, and G3. Case C1 is Gen 1 and produces cases C2, and C3 in generation 2. In Gen 3, C2 produces C4 and C5, and C3 produces C7. The chain ends in Gen 3. The chain size of C1 is 7 (that is the sum of all the blue circles, representing cases) and the length is 3 (maximum number of generations reached by C1's chain)."}
```

Comment:
> The generations are labelled "Gen X" in the figure and "GX" here - suggest to streamline. Also there's no C7 - is the image old version of the figure perhaps?

Thanks for catching that. Yes, I updated the figure later on and forgot to update the text. See change in 6161d8350f15eb4b75c0171a71b01914ded2d428.

> For the outbreak size / length, bear in mind that the index case is not counted (so the figure is really size 5 / length 2).

I think we do count the index case. See https://github.com/epiverse-trace/bpmodels/blob/main/R/simulate.r#L190-L195. `stat_track` starts off as 1 for each seeded case and is incremented based on `size` or `length`. Using bpmodels code because this part was converted into a helper function here so will require navigating between two places.

Reply:
Ah yes you're right, sorry.

Reply:
Should we not count the seeded cases? We could just subtract 1 from the resulting vector.

Reply:
I think you could argue either way - I find it clearer to include them i.e. as it is at the moment. But perhaps there's an opportunity here to clarify this here, i.e. add "(including the index case)" somewhere in the sentence before the fig. caption (l.51).

Reply:
I've clarified that here 8127f57b0e82f4fe4ae2990968f22a1783295be3.